src/backend-api/Controllers/UpdateController.cs

```diff
@@ -229,6 +227,196 @@ public IActionResult EditProgram([FromBody] PostProgramInputModel input, [FromRo
+         *           "LastName": String,
+         *           "HireDate": DateTime,
+         *           "Role": String,
+         *           "DepartmentID": int,
+         *           "IsAdmin" : bool
+         *       },
+         *       "HardwareAssigned": [
+         *            {
+         *               "Type": String,
+         *               "ID": int
+         *           }
+         *       ] for all the hardware assigned (this arraylist)
+         *       "ProgramAssigned": [
+         *           {
+         *               "ID": int
+         *           },
+         *       ] for all the programs assigned (this arraylist)
+         *       "HardwareUnassigned": [
+         *           {
+         *               "Type": String,
+         *               "ID": int
+         *           }
+         *       ] for all the hardware unassigned (this arraylist)
+         *       "ProgramUnassigned": [
+         *           {
+         *               "ID": int
+         *           }
+         *       ] for all the programs unassigned (this arraylist)
+         *   }
+         */
+
+        [HttpPut]
+        [Route("Employee")]
+        public IActionResult EditEmployee([FromBody] EditEmployeeInputModel input)
+        {
+            // finding the employee from the given employeeId
+            var emp = _context.Employee.Find(input.Employee.EmployeeId);
+
+            // finding the authID connected to this employee from adGuid
+            var authIdEmp = _context.AuthIdserver.Where(x => x.ActiveDirectoryId == emp.Adguid).FirstOrDefault();
+
+            // making sure that neither of these are null. These are both created when an employee is created
+            if (emp != null && authIdEmp !=null)
+            {
+                try
+                {
+                    // updating the various fields of the current employee
+                    emp.HireDate = input.Employee.HireDate;
+                    emp.FirstName = input.Employee.FirstName;
+                    emp.LastName = input.Employee.LastName;
+                    emp.Role = input.Employee.Role;
+                    emp.DepartmentID = input.Employee.DepartmentID;
+                    _context.Employee.Update(emp);
+
+                    // updating the current isAdmin for the authIdServer connected to the employee
+                    authIdEmp.IsAdmin = input.Employee.IsAdmin;
+                    _context.AuthIdserver.Update(authIdEmp);
+
+                    _context.SaveChanges();
+
+                    // checking if any hardware is to be assigned
+                    if (input.HardwareAssigned != null)
+                    {
+                        // loop through hardware and depending on what type the hardware is, then add the hardware to the specific table. 
+                        // also in the loop, update the hardware history whether the it was assigned or unassigned  
+                        foreach (var hardware in input.HardwareAssigned)
```

Comment:
Won't this create a new `HardwareHistory` row every time an employee is updated even if the hardware was already assigned to the employee?

Reply:
Or is the `HardwareAssigned` array in the input only going to contain new hardware that is assigned and not hardware currently assigned?

Reply:
According to the front-end, it should only contain new hardware that is assigned